mongo_connector/connector.py

```diff
@@ -315,7 +315,7 @@ def run(self):
                 for shard_doc in main_conn['config']['shards'].find():
                     shard_id = shard_doc['_id']
                     if shard_id in self.shard_set:
-                        if not self.shard_set[shard_id].running:
+                        if not (self.shard_set[shard_id].running and self.shard_set[shard_id].is_alive()):
```

Comment:
Could you move the "self.shard_set[shard_id].is_alive()" to the following line, so that lines don't exceed 80 characters in length?

    if not (self.shard_set[shard_id].running and
            self.shard_set[shard_id].is_alive()):
        LOG.error(...)

or

    shard_thread = self.shard_set[shard_id]
    if not (shard_thread.running and shard_thread.is_alive()):
        LOG.error(...)